Share Extension/ShareViewController.swift

```diff
@@ -7,48 +7,32 @@ class ShareViewController: UIViewController {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
-        // Get the attachment
         let content = extensionContext?.inputItems.first as? NSExtensionItem
         guard let attachment = content?.attachments?.first as? NSItemProvider else {
             close()
             return
         }
 
-        // We only accept OPML files, and they need to conform to data identifier
-        // If we ever want to accept other files we need to change the Share Extension
-        // Info.plist
-        if attachment.hasItemConformingToTypeIdentifier(UTType.data.identifier) {
-
-            // Request the OPML file URL
-            attachment.loadItem(forTypeIdentifier: UTType.data.identifier, options: nil) { [weak self] data, error in
-                guard let url = data as? URL else {
-                    self?.close()
-                    return
-                }
-
-                // Convert the file to Data
-                guard let opmlData = try? Data(contentsOf: url) else {
-                    self?.close()
-                    return
-                }
-
-                self?.close()
-
-                // Redirect to Pocket Casts sharing the OPML data encoded in Base64
-                self?.redirectToHostApp(opmlData.base64EncodedString())
-            }
+        // We accept 3 types of files: audio, movie and opml
+        var identifier: String = ""
+        if attachment.hasItemConformingToTypeIdentifier(UTType.audio.identifier) {
+            identifier = UTType.audio.identifier
+        } else if attachment.hasItemConformingToTypeIdentifier(UTType.movie.identifier) {
+            identifier = UTType.movie.identifier
+        } else if attachment.hasItemConformingToTypeIdentifier(UTType.data.identifier) {
+            identifier = UTType.data.identifier
         }
+        loadFile(from: attachment, identifier: identifier)
```

Comment:
We can simplify that a lot by doing this:

```suggestion
        let acceptedTypes: [UTType] = [.audio, .movie, .data]

        if let type = acceptedTypes.first(where: { attachment.hasItemConformingToTypeIdentifier($0.identifier) }) {
            loadFile(from: attachment, identifier: type.identifier)
        }
```

Alternatively, you can move `acceptedTypes` to be a class property.